perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -2771,19 +2784,25 @@ public Group createGroupUnion(PerunSession sess, Group resultGroup, Group operan
 		groupsManagerImpl.saveGroupRelation(sess, resultGroup, operandGroup, parentFlag);
 
 		// do the operation logic
-		processRelationMembers(sess, resultGroup, getGroupMembers(sess, operandGroup), operandGroup.getId(), true);
+		try {
+			addRelationMembers(sess, resultGroup, getGroupMembers(sess, operandGroup), operandGroup.getId());
+		} catch (WrongReferenceAttributeValueException | AlreadyMemberException | WrongAttributeValueException e ) {
+			throw new InternalErrorException("Some operation was not allowed during processing members relations", e);
+		} catch (GroupNotExistsException e) {
+			throw new InternalErrorException("Some group does not exists...", e);
+		}
```

Comment:
Is there any reason to throw InternalErrorException instead of other exceptions there?

Reply:
I was inspired by method moveGroup where it was solved the same way for the old processRelationMembers method, so i thought we followed this logic. What would be the best solution here? Not to catch the exceptions at all (as in old code over here) or to throw them but not mask them behind InternalErrorException?

Reply:
I think (if it is possible) to throw it out without change.